src/main/java/uk/gov/justice/digital/hmpps/keyworker/migration/Movement.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.digital.hmpps.keyworker.migration
+
+import uk.gov.justice.digital.hmpps.keyworker.model.DeallocationReason
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+data class Movement(
+  val offenderNo: String,
+  val movementDate: LocalDate,
+  val movementTime: LocalTime,
+  val fromAgency: String,
+  val toAgency: String,
+  val movementType: String,
+  val movementReasonCode: String,
+  val directionCode: String,
+) {
+  val occurredAt: LocalDateTime = lazy { LocalDateTime.of(movementDate, movementTime) }.value
+  val deallocationReason: DeallocationReason? =
+    lazy {
+      when (movementType) {
+        "TRN" if (directionCode == "OUT") -> DeallocationReason.TRANSFER
```

Comment:
I think we need to do the is prison check for transfers to match the automatic deallocation code